[snyk] Add snyk2nvd converter

n1chre opened the pull request:
This commit adds a snyk2nvd converter. The tool can download the feed
from snyk.io and convert it to NVD format.

Test: go build && go test

skogtwin commented on providers/snyk/converter/converter.go:
```diff
@@ -0,0 +1,426 @@
+	return out
+}
+
+type baseSnykItem struct {
```
This should be in schema.

skogtwin commented on providers/snyk/jsonschema/jsonschema.go:
```diff
@@ -0,0 +1,174 @@
+// Package jsonschema was auto-generated.
+// Command: jsonschema2go -gofmt -gopkg jsonschema -goptr snyk.json
+package jsonschema
```
let's replace all specific structs with a generic one from above.

skogtwin commented on providers/snyk/jsonschema/jsonschema.go:
and make this `map[sting]baseSnykItem`

skogtwin commented on providers/snyk/converter/converter.go:
see comments below/above, there should be just one of these.

n1chre commented:
Rewrote main per @skogtwin 's comments

n1chre commented:
Resolved all comments @skogtwin 
What was changed:
- simplified schema: everything is in the same format, so dont use SnykGolang, SnykJava, ... just use SnykAdvisory; Snyk is now a map[language][]*Advisories
- converters: only need to convert this one, not do any shenanigans with converting first to basesnyk item and then convert that. I didn't touch the logic, just deleted dupllication
